manifests/service.pp

```diff
@@ -5,23 +5,24 @@
   $manage_oddjobd = $sssd::manage_oddjobd,
   $service_ensure = $sssd::service_ensure,
 ) {
-  service { $sssd_service:
-    ensure     => $service_ensure,
-    enable     => true,
-    hasstatus  => true,
-    hasrestart => true,
-  }
```

Comment:
The puppetlabs/apache module at least has the ```hasrestart``` set to ```true``` by default.
https://github.com/puppetlabs/puppetlabs-apache/blob/master/manifests/service.pp#L42

https://docs.puppet.com/puppet/latest/reference/type.html#service-attribute-hasrestart
According to the doc these have explicit defaults and are not determined by the provider.
```hasrestart``` defaults to ```false```, so we definitely want to override that and set it to ```true```, since we know that the sssd service has a restart action on all supported platforms and it is better than doing a start then stop.

The ```hasstatus``` defaults to ```true```, which is what we were setting it to, so i'm less concearned about that being set...